api/v1beta1/placementapi_webhook.go

```diff
@@ -175,3 +175,25 @@ func ValidateDefaultConfigOverwrite(
 	}
 	return errors
 }
+
+// SetDefaultRouteAnnotations sets HAProxy timeout values of the route
+func (spec *PlacementAPISpecCore) SetDefaultRouteAnnotations(annotations map[string]string) {
+	const haProxyAnno = "haproxy.router.openshift.io/timeout"
+	// Use a custom annotation to flag when the operator has set the default HAProxy timeout
+	// With the annotation func determines when to overwrite existing HAProxy timeout with the APITimeout
+	const placementAnno = "api.placement.openstack.org/timeout"
+	valPlacementAPI, okPlacemenAPI := annotations[placementAnno]
+	valHAProxy, okHAProxy := annotations[haProxyAnno]
+	// Human operator set the HAProxy timeout manually
+	if !okPlacemenAPI && okHAProxy {
+		return
+	}
+	// Human operator modified the HAProxy timeout manually without removing the Placemen flag
+	if okPlacemenAPI && okHAProxy && valPlacementAPI != valHAProxy {
+		delete(annotations, placementAnno)
+		return
+	}
+	timeout := fmt.Sprintf("%ds", spec.APITimeout)
+	annotations[placementAnno] = timeout
+	annotations[haProxyAnno] = timeout
+}
```

Comment:
This feels sensible. Thanks for taking care of the possible conflicts.

Reply:
I'm wondering if the other operators using the same logic. At least they should. And then This function can be abstracted to lib-common as it only need the annotations param and a service specific annotation param but it does not depend on the spec at all.

Reply:
iirc, neutron was the first one adding this https://github.com/openstack-k8s-operators/neutron-operator/blob/main/api/v1beta1/neutronapi_webhook.go. we have functional tests as part of the openstack-operaotr https://github.com/openstack-k8s-operators/openstack-operator/blob/main/tests/functional/ctlplane/openstackoperator_controller_test.go#L563

Reply:
> I'm wondering if the other operators using the same logic. At least they should. And then This function can be abstracted to lib-common as it only need the annotations param and a service specific annotation param but it does not depend on the spec at all.

yes all operators use same logic so it can be move to lib-common but maybe as a follow up?

Reply:
follow up works for me